src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/failover/AuroraTopologyService.java

```diff
@@ -257,33 +253,48 @@ private boolean gatherPerfMetrics(PropertySet props) {
    */
   private ClusterTopologyInfo processQueryResults(ResultSet resultSet)
       throws SQLException {
-    int writerCount = 0;
 
     List<HostInfo> hosts = new ArrayList<>();
+    List<HostInfo> writers = new ArrayList<>();
     while (resultSet.next()) {
       if (!WRITER_SESSION_ID.equalsIgnoreCase(resultSet.getString(FIELD_SESSION_ID))) {
         hosts.add(createHost(resultSet));
         continue;
       }
 
-      if (writerCount == 0) {
-        // store the first writer to its expected position [0]
-        hosts.add(
-            FailoverConnectionPlugin.WRITER_CONNECTION_INDEX,
-            createHost(resultSet));
-      } else {
-        // append other writers, if any, to the end of the host list
-        hosts.add(createHost(resultSet));
-      }
-      writerCount++;
+      writers.add(createHost(resultSet));
     }
 
+    int writerCount = writers.size();
+
     if (writerCount == 0) {
       this.log.logError(Messages.getString("AuroraTopologyService.3"));
       hosts.clear();
+    } else if (writerCount == 1) {
+      // Store the first writer to its expected position [0]
+      hosts.add(FailoverConnectionPlugin.WRITER_CONNECTION_INDEX, writers.get(0));
+    } else {
+      List<HostInfo> previousTopology = topologyCache.get(this.clusterId);
+      HostInfo oldWriter = previousTopology == null ?
+          null : previousTopology.get(FailoverConnectionPlugin.WRITER_CONNECTION_INDEX);
+
+      // There is potentially a stale record. Do not add it to the hosts list.
+      if (writerCount == 2
```

Comment:
And if it's more than 2 writers found? Like few stale writer records. Is it possible to process all stale records and keep the latest one?